app/scripts/services/authorization.js

```diff
@@ -0,0 +1,169 @@
+      this.canCreateSomeResource = false;
+      this.wasCanCreateSomeResourceSet = false;
+      this.projectNameRules = null;
+    }
+
+    AuthorizationService.prototype.reviewUserRules = function($scope) {
+      var self = this;
+      var projectName = (!_.isUndefined($scope.project)) ? $scope.project.metadata.name : null;
+      self.getProjectRules(projectName).then(
+        function(userRulesLoaded) {
+          if (userRulesLoaded) {
+            $scope.canI = self.setPageRules($scope.canI);
+          } else {
+            $scope.alerts["rules"] = {
+              type: "error",
+              message: "Failed to load users rules."
+            };
+          }
+        });
+    };
+
+    // Method to retrieve user rules. Rules will be namespaced if namespace is provided. If not,
+    // non-namespaced rules will be retrieved(eg. projectrequest).
+    AuthorizationService.prototype.getProjectRules = function(projectName) {
+      var self = this;
+      var deferred = $q.defer();
+      if (!self.rules || self.forceReload || self.projectNameRules !== projectName) {
+        // Making an API call to get the rules if:
+        // - no rules are cached
+        // - when need to force update the rules(eg. switching from projects page to project overview)
+        // - when switching between projects
+        $log.debug("AuthorizationService, loading user rules");
+        if (self.forceReload) {
+          self.setForceReload(false);
+        }
+        if (!self.intervalTimer) {
+          // Force rules reload every 3 minutes
+          self.intervalTimer = $interval(function () {
+            self.setForceReload(true);
+          }, 180000);
+        }
+        self.projectNameRules = projectName;
+        var object = {kind: "SelfSubjectRulesReview",
+                      apiVersion: "v1"
+                    };
+        DataService.create('selfsubjectrulesreviews', null, object, {namespace: projectName}).then(
+          function(data) {
+            self.rules = data.status.rules;
+            deferred.resolve(true);
+          }, function() {
+            self.rules = null;
+            deferred.resolve(false);
+        });
+      } else {
+        // Using cached data.
+        $log.debug("AuthorizationService, using cached rules");
+        deferred.resolve(true);
+      }
+      return deferred.promise;
+    };
+
+    // Will traverse through all the user rules and determine what actions on the current page is user able to do.
+    // In case no page rules are set, function will determine whether user can create some resources on the page
+    // to know if the 'Add to Project' btn should be shown.
+    AuthorizationService.prototype.setPageRules = function(pageRules) {
+      var self = this;
+      var canCreate = false;
+
+      if (pageRules) {
+        _.each(this.rules, function(rule) {
+          _.each(pageRules, function(verbs, resource) {
+            if (_.indexOf(rule.resources, resource) !== -1) {
```

Comment:
yeah, forgot about this lodash function. Will have to tattoo it on forearm :smiley: